include/zephyr/drivers/flash.h

```diff
@@ -38,6 +38,26 @@ struct flash_pages_layout {
 };
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
+
+#define FLASH_DEV_INFOWORD(dev) (((struct flash_driver_api *)dev->api)->infoword)
+
+#if IS_ENABLED(CONFIG_FLASH_DEVICE_REQUIRES_ERASE) && !IS_ENABLED(CONFIG_FLASH_DEVICE_HAS_NO_ERASE)
+#define FLASH_ERASE_REQUIRED(dev) (true)
+#elif !IS_ENABLED(CONFIG_FLASH_DEVICE_REQUIRES_ERASE) && \
+	IS_ENABLED(CONFIG_FLASH_DEVICE_HAS_NO_ERASE)
+#define FLASH_ERASE_REQUIRED(dev) (false)
+#else
+#define FLASH_ERASE_REQUIRED(dev) (FLASH_DEV_INFOWORD(dev).erase_required == 1)
+#endif
+
+#if IS_ENABLED(CONFIG_FLASH_DEVICE_HAS_ERASE) && IS_ENABLED(CONFIG_FLASH_DEVICE_HAS_NO_ERASE)
```

Comment:
Why do we need these macros? Just set the flag in the infoword and read it directly, its simple and specific to each device

Reply:
Here is an example, that is for ZSAI but the macros are basically taken from there, https://github.com/zephyrproject-rtos/zephyr/pull/64727/commits/2ad4cc60e6744c71b0fdbe923e4f7b3e2735d3e3
that shows how the macros are in use when code may be accessing both types of devices.